### PR TITLE
fix: disable copybook analysis only on extended document

### PR DESCRIPTION
**Pull request by zacanbrcom**
Hotfix that address an issue introduce by #441 
In particular when document is in did_change mode, copybooks analysis needs to be skipped ONLY if the document is an extended document.

### CODE REVIEW

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/DocumentExtensionTests.java [hunk 1]
```diff
@@ -82,11 +82,11 @@ private void checkExtensionMatches(String extension) {
     fireDidOpen(extension, uri);
     verify(engine, timeout(10000).times(1)).analyze(uri, TEXT, CopybookProcessingMode.ENABLED);
 
-    when(engine.analyze(uri, TEXT, CopybookProcessingMode.DISABLED))
+    when(engine.analyze(uri, TEXT, CopybookProcessingMode.ENABLED))
         .thenReturn(AnalysisResult.empty());
     fireDidChange(uri);
     verify(engine, timeout(10000))
-        .analyze(uri, INCORRECT_TEXT_EXAMPLE, CopybookProcessingMode.DISABLED);
+        .analyze(uri, INCORRECT_TEXT_EXAMPLE, CopybookProcessingMode.SKIP);
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/DocumentExtensionTests.java:89]: Shouldn't the modes here be the same?

File: com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/CopybookProcessingMode.java [hunk 2]
```diff
@@ -14,32 +14,40 @@
...
    * @param uri of the document opened in the editor by the client
-   * @param textDocSyncType define the document sync mode that could be [{@link
-   *     TextDocumentSyncType#DID_OPEN}|{@link TextDocumentSyncType#DID_CHANGE}]
+   * @param textDocumentSyncType the sync type of the document [{@link
+   *     TextDocumentSyncType#DID_OPEN} or {@link TextDocumentSyncType#DID_CHANGE}]
    * @return A value of that could be [{@link CopybookProcessingMode#ENABLED}|{@link
-   *     CopybookProcessingMode#DISABLED}]
+   *     CopybookProcessingMode#DISABLED}] if the document is an extended document.
    */
   public static CopybookProcessingMode getCopybookProcessingMode(
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/CopybookProcessingMode.java:37]: In fact, you may pass sync type as an enum value, not as a string, and just compare them by link equity

Comment (zacanbrcom) [reply com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/CopybookProcessingMode.java:37]: right 👍

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/DocumentExtensionTests.java (hunk 1 above)
Comment (zacanbrcom) [reply com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/DocumentExtensionTests.java:89]: got it - ok, I was focused on the verify, fixed

**Comment by zacanbrcom** [comment]
thanks guys!

**Pull request merged**